api/nodes/views.py

```diff
@@ -1149,7 +1159,17 @@ def get_queryset(self):
         if not hasattr(node_addon, 'get_folders'):
             raise EndpointNotImplementedError('Endpoint not yet implemented for this addon')
 
-        return node_addon.get_folders(path=path, folder_id=folder_id)
+        #  Convert v1 errors to v2 as much as possible.
+        try:
+            return node_addon.get_folders(path=path, folder_id=folder_id)
+        except InvalidAuthError:
+            raise NotAuthenticated('This add-on could not be authenticated.')
+        except HTTPError as exc:
+            v2_exception = HTTP_CODE_MAP.get(exc.code)
+            if v2_exception:
+                raise v2_exception
+            else:
```

Comment:
Suuuuper minor: an `else` isn't necessary if you're raising an exception in the `if`, i.e.
```
v2_exception = HTTP_CODE_MAP.get(exc.code)
if v2_exception:
	raise v2_exception
raise ServiceUnavailableError(...)
```